### PR TITLE
[App] Add display name property to the work

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR enables to customize the name of the work on the LAI platform:

image

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: src/lightning_app/core/work.py [hunk 1]
```diff
@@ -119,7 +119,16 @@ def __init__(
...
-        self._state = {"_host", "_port", "_url", "_future_url", "_internal_ip", "_restarting", "_cloud_compute"}
+        self._state = {
+            "_host",
+            "_port",
+            "_url",
+            "_future_url",
+            "_internal_ip",
+            "_restarting",
+            "_cloud_compute",
+            "_display_name",
```
Comment (awaelchli) [inline src/lightning_app/core/work.py:130]: @tchaton Is the display name meant to be changeable? Will the UI update the name when it changes in the state?

File: src/lightning_app/core/work.py [hunk 2]
```diff
@@ -207,6 +217,19 @@ def name(self):
...
+    def display_name(self):
+        """Returns the display name of the LightningWork in the cloud.
+
+        The display name needs to set before the run method of the work is called.
+        """
+        return self._display_name
+
+    @display_name.setter
+    def display_name(self, display_name: str):
+        """Returns the display name of the LightningWork in the cloud."""
```
Comment (hhsecond) [inline src/lightning_app/core/work.py:230]:
```suggestion
        """Sets the display name of the LightningWork in the cloud."""
```

File: src/lightning_app/core/work.py [hunk 3]
```diff
@@ -207,6 +217,19 @@ def name(self):
...
 
+    @property
+    def display_name(self):
+        """Returns the display name of the LightningWork in the cloud.
+
+        The display name needs to set before the run method of the work is called.
+        """
+        return self._display_name
+
+    @display_name.setter
```
Comment (hhsecond) [inline src/lightning_app/core/work.py:228]: Continuing to what Adrian mentioned above, we shouldn't allow setting the name if it is after the run method started

**Review by hhsecond — APPROVED** [review]
Looks good overall. One question+change and another nit on the doc

File: src/lightning_app/core/work.py (hunk 2 above)
Comment (tchaton) [reply src/lightning_app/core/work.py:230]: Done !

File: src/lightning_app/core/work.py (hunk 1 above)
Comment (tchaton) [reply src/lightning_app/core/work.py:130]: You are right ! Added an exception if the work has already started.

File: src/lightning_app/core/work.py (hunk 3 above)
Comment (tchaton) [reply src/lightning_app/core/work.py:228]: Answered above.

**Pull request merged**